testdata/sqllogictests/datatypes.slt

```diff
@@ -27,12 +27,13 @@ select distinct arrow_typeof(f) from test
 ----
 FixedSizeList(Field { name: "item", data_type: Int64, nullable: true, dict_id: 0, dict_is_ordered: false, metadata: {} }, 2) 
 
-
 # array functions don't work on fixed size lists
 statement error
 select array_element(1, f) from test;
 
-statement error 
+# but this one does work? 
+# TODO: should this be allowed?
+statement ok
 select array_append(f, 3) from test;
```

Comment:
what more should we do here before merging

Reply:
I don't think mutating fixed sized arrays should be allowed. If it's a **fixed size**, then `array_append` doesn't seem like it should be valid. I'm not sure what we want to do with this one yet, maybe upstream something to datafusion? or monkeypatch the function to not work on fsl's?

Reply:
I think we should just leave this for now. If it works in datafusion, i think it's fine for it to work in our system too. It just implicitly coerces the fsl to a regular list. 

I also think this a pretty rarely used function (especially on FSL's). I'll open up a separate issue to track this, but I don't think we need to act on it now.